packages/modules/payment/src/services/payment-module.ts

```diff
@@ -763,27 +773,44 @@ export default class PaymentModuleService
   protected async capturePaymentFromProvider_(
     payment: InferEntityType<typeof Payment>,
     capture: InferEntityType<typeof Capture> | undefined,
-    isFullyCaptured: boolean,
+    options: {
+      isFullyCaptured?: boolean
+      isCaptured?: boolean
+    } = {},
     @MedusaContext() sharedContext: Context = {}
   ) {
-    const paymentData = await this.paymentProviderService_.capturePayment(
-      payment.provider_id,
-      {
-        data: payment.data!,
-        context: {
-          idempotency_key: capture?.id,
-        },
-      }
-    )
+    if (!options.isCaptured) {
+      const paymentData = await this.paymentProviderService_.capturePayment(
+        payment.provider_id,
+        {
+          data: payment.data!,
+          context: {
+            idempotency_key: capture?.id,
+          },
+        }
+      )
 
-    await this.paymentService_.update(
-      {
-        id: payment.id,
-        data: paymentData.data,
-        captured_at: isFullyCaptured ? new Date() : undefined,
-      },
-      sharedContext
-    )
+      await this.paymentService_.update(
+        {
+          id: payment.id,
+          data: paymentData.data,
+          captured_at: options.isFullyCaptured ? new Date() : undefined,
+        },
+        sharedContext
+      )
+    } else if (options.isFullyCaptured && !payment.captured_at) {
+      /**
+       * In the case of auto capture, we need to update the payment to set the captured_at date
+       * only if fully captured.
+       */
+      await this.paymentService_.update(
+        {
+          id: payment.id,
+          captured_at: options.isFullyCaptured ? new Date() : undefined,
```

Comment:
nit: we are already checking for a full capture in the else-if
```suggestion
          captured_at: new Date(),
```